spikeinterface/toolkit/postprocessing/principal_component.py

```diff
@@ -353,26 +357,53 @@ def _fit_by_channel_local(self):
 
         # there is one PCA per channel for independent fit per channel
         pca_model = [IncrementalPCA(n_components=p['n_components'], whiten=p['whiten']) for _ in channel_ids]
-
+        
+        mode = p["mode"]
+        pca_model_files = []
+        for chan_ind, chan_id in enumerate(channel_ids):
+            pca = pca_model[chan_ind]
+            pca_model_file = self.extension_folder / f"pca_model_{mode}_{chan_id}.pkl"
+            with pca_model_file.open("wb") as f:
+                pickle.dump(pca, f)
+            pca_model_files.append(pca_model_file)
+        
         # fit
-        for unit_id in unit_ids:
+        if verbose:
+            units_loop = tqdm(np.arange(len(unit_ids)), desc="Fitting PCA")
+        else:
+            units_loop = np.arange(len(unit_ids))
+
+        for ui in units_loop:
+            unit_id = unit_ids[ui]
             wfs = we.get_waveforms(unit_id)
             if len(wfs) < p['n_components']:
                 continue
+            # try to paralleliza this
+            if n_jobs in (0, 1):
+                for chan_ind, chan_id in enumerate(channel_ids):
+                    pca = pca_model[chan_ind]
+                    pca.partial_fit(wfs[:, :, chan_ind])
+            else:
+                Parallel(n_jobs=n_jobs)(delayed(partial_fit_one_channel)(pca_model_files[chan_ind], wfs[:, :, chan_ind])
+                                                        for chan_ind in np.arange(len(channel_ids)))
```

Comment:
SO here we only parral along channels? I thioght you want to parralelise accros units ? no ?

Reply:
No here we can only paralleliza across channels because we partial fit each channel separately, but with all units!